src/esrs/postgres/mod.rs

```diff
@@ -179,55 +178,66 @@ impl<
-        let store_event_result: Result<PgQueryResult, Err> = sqlx::query(self.queries.insert())
-            .bind(event_id)
-            .bind(aggregate_id)
-            .bind(Json(&event))
-            .bind(occurred_on)
-            .bind(sequence_number)
-            .execute(&mut *connection)
-            .await
-            .map_err(|error| error.into());
-
-        let rebuild_result: Result<StoreEvent<Evt>, Err> = match store_event_result {
-            Ok(_) => {
-                let store_event: StoreEvent<Evt> = StoreEvent {
-                    id: event_id,
-                    aggregate_id,
-                    payload: event,
-                    occurred_on,
-                    sequence_number,
-                };
-
-                self.project_event(&store_event, &mut connection)
-                    .await
-                    .map(|()| store_event)
+
+        let events: Vec<_> = events
+            .into_iter()
+            .map(|e| (Uuid::new_v4(), e))
+            .zip(starting_sequence_number..)
+            .collect();
+
+        for ((event_id, event), sequence_number) in events.iter() {
+            let result = sqlx::query(self.queries.insert())
+                .bind(event_id)
+                .bind(aggregate_id)
+                .bind(Json(event))
+                .bind(occurred_on)
+                .bind(sequence_number)
+                .execute(&mut *connection)
+                .await;
+
+            if let Err(err) = result {
+                self.rollback(connection).await?;
+                return Err(err.into());
             }
-            Err(error) => Err(error),
-        };
+        }
 
-        match rebuild_result {
-            Ok(event) => {
-                self.commit(connection).await?;
+        let store_events: Vec<_> = events
+            .into_iter()
+            .map(|((event_id, event), sequence_number)| StoreEvent {
+                id: event_id,
+                aggregate_id,
+                payload: event,
+                occurred_on,
+                sequence_number,
+            })
+            .collect();
+
+        for store_event in store_events.iter() {
+            let result = self.project_event(store_event, &mut connection).await;
+
+            if let Err(err) = result {
+                self.rollback(connection).await?;
+                return Err(err);
+            }
+        }
 
-                for policy in &self.policies {
-                    policy.handle_event(&event, &self.pool).await?
-                }
+        self.commit(connection).await?;
 
-                Ok(event)
-            }
-            Err(err) => {
-                self.rollback(connection).await?;
-                Err(err)
+        // REVIEW: This implies that potentially half of the policies would trigger, then one fails, and the rest wouldn't.
+        // potentially we should be returning some other kind of error, that includes the errors from any failed policies?
```

Comment:
this would be really helpful, even if probably there would be nothing to do, beside having a better error logging. Right?

Reply:
I think so - I'll look at raising a separate issue for this later, since this was the previous behaviour - so may be better to deal with it separately 😄

Reply:
yes, so that we can discuss it too :)